app/src/main/java/com/github/libretube/ui/adapters/SearchChannelAdapter.kt

```diff
@@ -0,0 +1,182 @@
+import com.github.libretube.api.obj.StreamItem
+import com.github.libretube.constants.IntentData
+import com.github.libretube.databinding.ChannelRowBinding
+import com.github.libretube.databinding.PlaylistsRowBinding
+import com.github.libretube.databinding.VideoRowBinding
+import com.github.libretube.enums.PlaylistType
+import com.github.libretube.extensions.formatShort
+import com.github.libretube.extensions.toID
+import com.github.libretube.helpers.ImageHelper
+import com.github.libretube.helpers.NavigationHelper
+import com.github.libretube.ui.adapters.callbacks.SearchCallback
+import com.github.libretube.ui.base.BaseActivity
+import com.github.libretube.ui.extensions.setFormattedDuration
+import com.github.libretube.ui.extensions.setWatchProgressLength
+import com.github.libretube.ui.extensions.setupSubscriptionButton
+import com.github.libretube.ui.sheets.ChannelOptionsBottomSheet
+import com.github.libretube.ui.sheets.PlaylistOptionsBottomSheet
+import com.github.libretube.ui.sheets.VideoOptionsBottomSheet
+import com.github.libretube.ui.viewholders.SearchViewHolder
+import com.github.libretube.util.TextUtils
+import kotlinx.serialization.encodeToString
+
+// TODO: Replace with SearchResultsAdapter when migrating the channel fragment to use Paging as well
+class SearchChannelAdapter : ListAdapter<ContentItem, SearchViewHolder>(SearchCallback) {
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): SearchViewHolder {
+        val layoutInflater = LayoutInflater.from(parent.context)
+
+        return when (viewType) {
+            0 -> SearchViewHolder(VideoRowBinding.inflate(layoutInflater, parent, false))
+            1 -> SearchViewHolder(ChannelRowBinding.inflate(layoutInflater, parent, false))
+            2 -> SearchViewHolder(PlaylistsRowBinding.inflate(layoutInflater, parent, false))
+            else -> throw IllegalArgumentException("Invalid type")
+        }
+    }
+
+    override fun onBindViewHolder(holder: SearchViewHolder, position: Int) {
+        val searchItem = currentList[position]
+
+        val videoRowBinding = holder.videoRowBinding
+        val channelRowBinding = holder.channelRowBinding
+        val playlistRowBinding = holder.playlistRowBinding
+
+        if (videoRowBinding != null) {
+            bindVideo(searchItem, videoRowBinding, position)
+        } else if (channelRowBinding != null) {
+            bindChannel(searchItem, channelRowBinding)
+        } else if (playlistRowBinding != null) {
+            bindPlaylist(searchItem, playlistRowBinding)
+        }
+    }
+
+    override fun getItemViewType(position: Int): Int {
+        return when (currentList[position].type) {
+            StreamItem.TYPE_STREAM -> 0
+            StreamItem.TYPE_CHANNEL -> 1
+            StreamItem.TYPE_PLAYLIST -> 2
+            else -> 3
+        }
+    }
+
+    private fun bindVideo(item: ContentItem, binding: VideoRowBinding, position: Int) {
+        binding.apply {
```

Comment:
```suggestion
        binding.apply {
            thumbnail.setImageDrawable(null)
```
This should fix that the thumbnails are changing during scrolling by showing the placeholder instead.

The same should be done for all other thumbnails and channel avatars, also for `bingPlaylist` and `bindChannel`.